pkg/cmd/cli/cmd/set/env.go

```diff
@@ -176,28 +179,54 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 
 	skipped := 0
 	for _, info := range infos {
-		ok, err := f.UpdatePodSpecForObject(info.Object, func(spec *kapi.PodSpec) error {
-			containers, _ := selectContainers(spec.Containers, containerMatch)
-			if len(containers) == 0 {
-				fmt.Fprintf(cmd.Out(), "warning: %s/%s does not have any containers matching %q\n", info.Mapping.Resource, info.Name, containerMatch)
+		ok, err := f.MutateEnvForObject(info.Object, func(obj interface{}) error {
```

Comment:
So here this should be
```go
ok, err := f.MutateEnvForObject(obj)
```
and in the factory you would break things down first to top level objects and then to the appropriate nested objects.